docs/developer/adr/0006-add-readonly-flags.rst

```diff
@@ -0,0 +1,65 @@
+ADR 0006: Add read-only flags
+=============================
+
+- Status: accepted
+- Deciders: Jan-Lukas, Neil, Owen, Simon
+- Date: 2021-05-05
+
+Context
+-------
+
+There are a number of contexts where values of variables are conceptually "broadcast" (not necessarily using an actual ``broadcast`` operation) and are thus shared (not to be confused with sharing from the shallow-copy mechanism):
+
+- Explicit ``broadcast`` operations.
+- Masks of dimension ``N-M`` if data is of dimension ``N``.
```

Comment:
Is this the wrong way round? Data `N-M` with mask of dimension `N`

Reply:
No, I mean, e.g., mask with `dims=['x']` and data `dims=['x',y']`.

Reply:
That's the way I think you intended it, but it doesn't read right: 
Masks of dimension ``N-M`` if data is of dimension ``N`` -> Masks of dimension ``N`` if data is of dimension ``N-M``

Reply:
I don't understand, if N and M are positive then `N-M < N`, so I want to refer to masks of dimension `N-M`.

Reply:
Ok. I think I understand what you trying to say here. I didn't read it right first time, It does make sense. I think your wording https://github.com/scipp/scipp/pull/1886#discussion_r626372730 might be better, but ok to Leave it as is.